File: app/src/main/java/com/example/podclassic/object/MusicList.java
package com.example.podclassic.object;

import com.example.podclassic.util.MediaStoreUtil;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;



public class MusicList {
    public String name;
    public Long id;
    public String artist;

    public int size;
    public int type = 0;

    public static final int TYPE_SINGER = 1;
    public static final int TYPE_ALBUM = 2;

    private ArrayList<Music> list;
    public ArrayList<Music> getList() {
        if (list == null) {
            if (type == TYPE_SINGER) {
                list = MediaStoreUtil.INSTANCE.getArtistMusic(name);
            } else if (type == TYPE_ALBUM) {
                list = MediaStoreUtil.INSTANCE.getAlbumMusic(name);
            }
        }
        return list;
    }

    public void setList(ArrayList<Music> list) {
        this.list = list;
    }

    public MusicList() { }

    public MusicList(String name, ArrayList<Music> list) {
        this.list = list;
        this.name = name;
    }

    public int getSize() {
        if (list != null && list.size() != 0) {
            return list.size();
        }
        return size;
    }

    @NotNull
    @Override
    public String toString() {
        return name + size;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (!(obj instanceof MusicList)) {
            return false;
        }
        MusicList musicList = (MusicList) obj;
        try {
            return name.equals(musicList.name);
        } catch (Exception e) {
            return false;
        }
    }

    @Override
    public int hashCode() {
        return name.hashCode();
    }

}

File: app/src/main/java/com/example/podclassic/bean/MusicList.kt
package com.example.podclassic.bean

import com.example.podclassic.util.MediaStoreUtil
import com.example.podclassic.util.PinyinUtil

data class MusicList(
    val title: String,
    val subtitle: String,
    val data: String,
    val id: Long? = 0L,
    val type: Int = TYPE_NONE,
    var list: ArrayList<Music>?
) : Comparable<MusicList> {

    companion object {
        const val TYPE_NONE = 0
        const val TYPE_ALBUM = 1
        const val TYPE_ARTIST = 2
        const val TYPE_FOLDER = 3
    }

    private var pinyin: String? = null
        get() {
            if (field == null) {
                field = PinyinUtil.getPinyin(title)
            }
            return field
        }

    class Builder {
        var title: String = ""
        var data: String = ""
        var subtitle : String = ""
        var id: Long? = null
        var type: Int = 0
        var list: ArrayList<Music>? = null
        fun build(): MusicList {
            return MusicList(title, subtitle, data, id, type, list)
        }
    }

    fun getMusicList(): ArrayList<Music> {
        if (list == null) {
            list = if (type == TYPE_ALBUM) {
                MediaStoreUtil.getAlbumMusic(title)
            } else {
                MediaStoreUtil.getArtistMusic(title)
            }
        }
        return list!!
    }

    override fun hashCode(): Int {
        return data.hashCode() * type
    }

    override fun equals(other: Any?): Boolean {
        if (other == null) {
            return false
        }
        if (other !is MusicList) {
            return false
        }
        return title == other.title && type == other.type && data == other.data
    }

    override fun compareTo(other: MusicList): Int {
        return pinyin!!.compareTo(other.pinyin!!)
    }
}